x-pack/platform/plugins/shared/lens/public/visualizations/partition/visualization.tsx

```diff
@@ -161,10 +166,9 @@ export const getPieVisualization = ({
 
   triggers: [VIS_EVENT_TO_TRIGGER.filter],
 
-  initialize(addNewLayer, state, mainPalette) {
-    if (state) {
-      return convertToRuntime(state);
-    }
+  initialize(addNewLayer, state, mainPalette, datasourceState) {
+    if (state) return convertToRuntimeState(state, datasourceState);
```

Comment:
I think it makes sense to deep copy here

Reply:
Fixed in 1e920004c857f71ac7f02b069cc9728edf5aca28